File: tests/xcalablemp/global-view/gmove/F/gmove_ata_4.f90
program gmove_ata_4

  character*25 tname
  call gmove_ata_2a1t_subcomm(tname)

end program

subroutine gmove_ata_2a1t_subcomm(tname)

character(*) tname
integer :: i,j
integer,parameter :: n=5
integer a(n,n),b(n,n)
integer xmp_node_num
!$xmp nodes p(4)
!$xmp nodes p1(2)=p(1:2)
!$xmp nodes p2(2)=p(3:4)
!$xmp template tx(n)
!$xmp template ty(n)
!$xmp distribute tx(block) onto p1
!$xmp distribute ty(block) onto p2
!$xmp align a(i,*) with tx(i)
!$xmp align b(*,i) with ty(i)

!$xmp loop (i) on tx(i)
do j=1,n
  do i=1,n
    a(i,j)=i+j
  end do
end do

!$xmp loop (j) on ty(j)
do j=1,n
  do i=1,n
    b(i,j)=0
  end do
end do

!$xmp gmove
b(1:n,1:n)=a(1:n,1:n)

ierr=0
!$xmp loop (i) on ty(i)
do j=1,n
  do i=1,n
    ierr=ierr+abs(b(j,i)-i-j)
  end do
end do

!$xmp reduction (max:ierr)
tname="gmove_ata_2a1t_subcomm"
call chk_int(tname, ierr)

end subroutine

File: tests/xcalablemp/global-view/gmove/C-Round/gmove_ata_4.c
#define NAMELEN 25
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

extern int chk_int(char name[], int ierr);

void gmove_ata_2a1t_subcomm(){

char name[NAMELEN]="gmove_bca_1a2t_b6";
int n=9;
double a[n][n],b[n][n];
#pragma xmp nodes p(4)
#pragma xmp nodes p1(2)=p(1:2)
#pragma xmp nodes p2(2)=p(3:4)
#pragma xmp template tx(0:n-1)
#pragma xmp template ty(0:n-1)
#pragma xmp distribute tx(block) onto p1
#pragma xmp distribute ty(block) onto p2
#pragma xmp align a[i][*] with tx(i)
#pragma xmp align b[*][i] with ty(i)

  int i,j,ierr;
  double err;

#pragma xmp loop (i) on tx(i)
  for(i=0;i<n;i++){
    for(j=0;j<n;j++){
      a[i][j]=i+j+2;
    }
  }

  for(i=0;i<n;i++){
#pragma xmp loop (j) on ty(j)
    for(j=0;j<n;j++){
      b[i][j]=0;
    }
  }

#pragma xmp gmove
  b[0:n][0:n]=a[0:n][0:n];

  err=0.0;
  for(i=0;i<n;i++){
#pragma xmp loop (j) on ty(j)
    for(j=0;j<n;j++){
      err=err+fabs(b[i][j]-i-j-2);
    }
  }

#pragma xmp reduction (MAX:err)
  ierr=err;
  chk_int(name, ierr);

}

int main(){

  gmove_ata_2a1t_subcomm();

  return 0;

}
